Realm.Shared/Session.cs

```diff
@@ -0,0 +1,40 @@
+﻿////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2016 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+using System;
+
+namespace Realms
+{
+    public enum SessionState
+    {
+        Authenticating,
+        Binding,
+        Bound,
+        Initial,
+        Stopped,
+        Unbound
+    }
+
+    public class Session
+    {
+        public SyncConfiguration Configuration { get; private set; }
+        public Uri ServerUri { get; private set; }
+        public SessionState State { get; private set; }
```

Comment:
Can we expose a `StateChanged` event or something similar? I mean - do we get notifications like that from Object Store?